### DISCUSSION TITLE
Access denied for user 'firefly'@'docker.ip' (using password: YES)

### DISCUSSION THREAD

**Discussion by siege801** [firefly-iii/firefly-iii#5528] [General]
Hello all,

I'm embarrassed to be stuck with this problem, because it seems like it should be an obvious thing to fix.

This is a new Firefly_iii installation that I'm trying to spin up. My docker-compose.yml section and the SQL section of the env file are below. The database logs are repeatedly saying:

`2022-01-12  9:48:51 50 [Warning] Access denied for user 'ff3User'@'docker.ip.add.ress' (using password: YES)`

Even if I enter the database docker container with `docker exec -it firefly_iii_db /bin/bash` I'm unable to log into mysql using the username and password from the yml and env files. (eg `mysql -u ff3User -p`). There is obviously something very obvious that I'm missing, but I can't figure it out. 

_NOTE: I set the title of this post to a more standardised database user (i.e. the default). But the docker-compose.yml and env files both definitely match (ff3User)._

# Service Definition in docker-compose.yml
`#------------------------------------------------------------------------------#
# FIREFLY CONFIGURATION
#------------------------------------------------------------------------------#        
services:
  app:
    container_name: firefly_iii
    image: fireflyiii/core:latest
    restart: always
    volumes:
      - /usr/share/docker/firefly_iii/firefly_iii_upload:/var/www/html/storage/upload
    env_file: firefly_iii.env
    ports:
      - 41211:8080
    depends_on:
      - db
  db:
    container_name: firefly_iii_db
    image: mariadb
    hostname: fireflyiiidb
    restart: always
    environment:
      - MYSQL_RANDOM_ROOT_PASSWORD=yes
      - MYSQL_USER=ff3User
      - MYSQL_PASSWORD=ff3Pass
      - MYSQL_DATABASE=ff3Dbase
    volumes:
      - firefly_iii_db:/var/lib/mysql
`

# Snip from firefly_iii.env
`DB_CONNECTION=mysql
DB_HOST=fireflyiiidb
DB_PORT=3306
DB_DATABASE=ff3Dbase
DB_USERNAME=ff3User
DB_PASSWORD=ff3Pass`

**Comment by JC5** [comment]
Did you start both containers and *then* changed the password?
